client-app/pages/loyalty/loyalty.vue

```diff
@@ -0,0 +1,456 @@
+  loadMyUserBalance,
+} = useUserBalance({ userId: currentUserId, storeId });
+
+const isMobile = breakpoints.smaller("md");
+const editingMode: Ref<boolean> = ref(false);
+const newOperation: Ref<InputRegisterPointsOperationType | null> = ref(null);
+const infiniteScrollEnabled = ref(false);
+const page = ref(1);
+const itemsPerPage = ref(20);
+const saveOperationLoading = ref(false);
+
+const title: ComputedRef<string> = computed(() => {
+  return editingMode.value
+    ? t("pages.loyalty.points_operations.operations_list.new_operation_title")
+    : t("pages.loyalty.points_operations.operations_list.operations_title");
+});
+
+const columns = ref<ITableColumn[]>([
+  {
+    id: "createdDate",
+    title: t("pages.loyalty.points_operations.operations_list.created_date_label"),
+    sortable: true,
+  },
+  {
+    id: "reason",
+    title: t("pages.loyalty.points_operations.operations_list.reason_label"),
+    sortable: true,
+  },
+  {
+    id: "amount",
+    title: t("pages.loyalty.points_operations.operations_list.amount_label"),
+    sortable: true,
+  },
+  {
+    id: "balance",
+    title: t("pages.loyalty.points_operations.operations_list.balance_label"),
+    sortable: true,
+  },
+]);
+
+async function onScrollModeChange() {
+  page.value = 1;
+  await loadOperations();
+}
+
+const onPageChange = async (newPage: number) => {
+  page.value = newPage;
+  await loadOperations();
+};
+
+// if address parameter is NULL, then adding a new address will open
+function openEditMode() {
+  const operation: InputRegisterPointsOperationType = {
+    userId: currentUserId,
+    storeId,
+    reason: "",
+    amount: 0,
+  };
+
+  newOperation.value = clone(operation);
+  editingMode.value = true;
+}
+
+function closeEditMode() {
+  newOperation.value = null;
+  editingMode.value = false;
+}
+
+function formatDate(value: string): string {
+  return value ? new Date(value).toLocaleString() : "";
+}
+
+async function saveOperation(operation: InputRegisterPointsOperationType): Promise<void> {
+  const { reason, amount } = operation;
+
+  saveOperationLoading.value = true;
+  await addPointsOperation(reason, Number(amount));
+  closeEditMode();
+  saveOperationLoading.value = false;
+
+  loadData();
+}
+
+onMounted(async () => {
+  await loadData();
+});
+
+async function loadData() {
+  await Promise.all([loadOperations(), loadMyUserBalance(false)]);
+}
+
+async function loadOperations() {
+  const searchParams: PointsOperationsSearchParams = {
+    userId: currentUserId,
+    storeId,
+    page: page.value,
+    itemsPerPage: itemsPerPage.value,
+    sort: "",
+  };
+
+  fetchMyPointsOperations(searchParams);
+}
+
+async function loadMoreOperations() {
+  if (page.value == pages.value) {
+    return;
+  }
+
+  const nextPage = page.value + 1;
+  page.value = nextPage;
+
+  const searchParams: PointsOperationsSearchParams = {
+    userId: currentUserId,
+    storeId,
+    page: nextPage,
+    itemsPerPage: itemsPerPage.value,
+    sort: "",
```

Comment:
Do you need to explicitly pass sort direction? I have no idea about how sorting works here.